src/main/java/org/littleshoot/proxy/extras/SelfSignedSslEngineSource.java

```
@@ -70,6 +98,83 @@ public SelfSignedSslEngineSource() {
+        		remoteServerCertInfo.set(CertificateAlgorithmId.NAME + "." + CertificateAlgorithmId.ALGORITHM, sigAlgid);
+
+        		CertificateExtensions extensions = (CertificateExtensions)remoteServerCertInfo.get(X509CertInfo.EXTENSIONS);
+        		Enumeration<Extension> elements = extensions.getElements();
+        		CertificateExtensions newExtensions = new CertificateExtensions();
+        		while(elements!=null && elements.hasMoreElements()){
+        			Extension ext = elements.nextElement();
+        			if(ext instanceof SubjectAlternativeNameExtension){
+        				newExtensions.set(SubjectAlternativeNameExtension.NAME, ext);
+        			}
+        		}
+
+        		//newExtensions.set(SubjectKeyIdentifierExtension.NAME, new SubjectKeyIdentifierExtension(new KeyIdentifier(caPublicKey).getIdentifier()));
+        		remoteServerCertInfo.set(X509CertInfo.EXTENSIONS, newExtensions);
+
+        		// Sign the new certificate
+        		newCert = new X509CertImpl(remoteServerCertInfo);
+        		newCert.sign(caPrivateKey, SIG_ALG_NAME);
+
+
+        		KeyStore serverKeyStore = KeyStore.getInstance(KEY_STORE_TYPE);
+        		serverKeyStore.load(null, PASSWORD.toCharArray());
+
+        		serverKeyStore.setCertificateEntry(ALIAS, newCert);
+        		serverKeyStore.setKeyEntry(ALIAS, caPrivateKey, PASSWORD.toCharArray(), new Certificate[] { newCert, caCert });
+
+        		final KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        		keyManagerFactory.init(serverKeyStore, PASSWORD.toCharArray());
+
+        		newSslContext.init(keyManagerFactory.getKeyManagers(), new TrustManager[] { new TrustEveryone() }, null);
+        		
+        		sslContextCache.put(sslContextKey, new SoftReference<SSLContext>(newSslContext));
+    		}
+    		
+    		return newSslContext.createSSLEngine();
+            
+    	}catch(Exception e){
+    		throw new RuntimeException("newSslEngine", e);
```

Can you throw a new exception type here, something like MitmSSLException, or something along those lines? I'm thinking that clients may want to catch that exception and do something with it, like retry without SSL/MITM, or display/log a different error message. (I anticipate doing this.)